src/confdb/confdb_setup.c

```diff
@@ -215,19 +219,14 @@ static int confdb_init_db(const char *config_file,
     }
     in_transaction = true;
 
-    /* Purge existing database, if we are reinitializing the confdb completely */
-    if (only_section == NULL) {
-        ret = confdb_purge(cdb);
-        if (ret != EOK) {
-            DEBUG(SSSDBG_FATAL_FAILURE,
-                "Could not purge existing configuration\n");
-            goto done;
-        }
+    ret = confdb_purge(cdb);
```

Comment:
There is a difference.
If any other ('sssd_*') process keeps config-db open, currently it will get updated content. If file gets deleted, they won't.
But maybe it's not important.